### PR TITLE
ci-automation/vm: build PXE if Equinix Metal is built

### PR DESCRIPTION
**Pull request by tormath1**
Signed-off-by: Mathieu Tortuyaux <mtortuyaux@microsoft.com>

## Testing done
Tested there: http://jenkins.infra.kinvolk.io:8080/job/container/job/vms/125/cldsv/

- [ ] Changelog entries added in the respective `changelog/` directory (user-facing change, bug fix, security fix, update)

### CODE REVIEW

File: ci-automation/vms.sh [hunk 1]
```diff
@@ -60,6 +60,13 @@ function vm_build() {
     echo "docker container rm -f '${vms_container}'" >> ci-cleanup.sh
 
     for format; do
+        # build automatically PXE for Equinix Metal.
+        if [[ "${format}" == "equinix_metal" ]]; then
+            ./run_sdk_container -n "${vms_container}" -C "${image_image}" \
+                -v "${vernum}" \
+                ./image_to_vm.sh --format pxe --board="${arch}-usr" \
+                    --from "${CONTAINER_IMAGE_ROOT}/${arch}-usr/latest"
+        fi
```
Comment (krnowak) [inline ci-automation/vms.sh:69]: I'd propose to run something like that before the `for` loop:

```sh
local has_packet=0
local has_pxe=0
for format; do
    if [[ "${format}" = 'packet' ]] || [[ "${format}" = 'equinix_metal' ]]; then has_packet=1; fi
    if [[ "${format}" = 'pxe' ]]; then has_pxe=1; fi
done
if [[ ${has_packet} -eq 1 ]] && [[ ${has_pxe} -eq 0 ]]; then set -- 'pxe' "${@}"; fi
```

Comment (krnowak) [reply ci-automation/vms.sh:69]: Or use `case`.

**Review by krnowak — APPROVED** [review]

**Comment by pothos** [comment]
I would like to do these changes to handle the conversion of platform names to the image formats we generate for them, I can file a new PR instead of touching this PR again

```
diff --git a/ci-automation/vms.sh b/ci-automation/vms.sh
index 3c7dd013..40977cbf 100644
--- a/ci-automation/vms.sh
+++ b/ci-automation/vms.sh
@@ -69,10 +69,24 @@ function vm_build() {
 
     [[ ${has_packet} -eq 1 ]] && [[ ${has_pxe} -eq 0 ]] && set -- 'pxe' "${@}"
 
-    for format; do
-        # keep compatibility with SDK scripts where "equinix_metal"
-        # remains unknown.
-        [ "${format}" = "equinix_metal" ] && format="packet"
+    # Convert platform names (also used to find the test scripts) to image formats they entail
+    formats="$*"
+    if echo "$formats" | tr ' ' '\n' | grep -q azure; then
+      formats=$(echo "$formats" | tr ' ' '\n' | sed '/azure.*/d')
+      formats+=" azure azure_gen2"
+    fi
+    if echo "$formats" | tr ' ' '\n' | grep -q '^vmware'; then
+      formats=$(echo "$formats" | tr ' ' '\n' | sed '/vmware.*/d')
+      formats+=" vmware vmware_insecure vmware_ova vmware_raw"
+    fi
+    if echo "$formats" | tr ' ' '\n' | grep -q -P '^(ami|aws)'; then
+      formats=$(echo "$formats" | tr ' ' '\n' | sed '/ami.*/d' | sed '/aws/d')
+      formats+=" ami ami_vmdk"
+    fi
+    # Keep compatibility with SDK scripts where "equinix_metal" remains unknown.
+    formats=$(echo "$formats" | tr ' ' '\n' | sed 's/equinix_metal /packet/g')
+
+    for format in ${formats}; do
         echo " ###################  VENDOR '${format}' ################### "
         ./run_sdk_container -n "${vms_container}" -C "${image_image}" \
             -v "${vernum}" \
```

**Comment by tormath1** [comment]
final test: http://jenkins.infra.kinvolk.io:8080/job/container/job/vms/125/cldsv/

**Pull request merged**